mkdocs_rss_plugin/util.py

```diff
@@ -495,24 +490,23 @@ def get_description_or_abstract(
                 in_page.markdown[:excerpt_separator_position],
                 output_format="html5",
             )
-        # If chars count is unlimited, use the html content
-        elif in_page.content and chars_count == -1:
-            if chars_count is None or len(in_page.content) < chars_count:
-                return in_page.content[:chars_count]
-        # Use markdown
-        elif in_page.markdown:
-            if chars_count is None or len(in_page.markdown) < chars_count:
-                return markdown.markdown(
-                    in_page.markdown[:chars_count], output_format="html5"
-                )
+        # Use first chars_count from the markdown
+        elif chars_count > 0 and in_page.markdown:
+            if len(in_page.markdown) <= chars_count:
+                return markdown.markdown(in_page.markdown, output_format="html5")
             else:
                 return markdown.markdown(
                     f"{in_page.markdown[: chars_count - 3]}...",
                     output_format="html5",
                 )
-        # Unlimited chars_count but no content is found, then return the description.
+        # No explicit description and no (or empty) abstract found
         else:
-            return description if description else ""
+            logger.warning(
+                f"No description set for page {in_page.file.src_uri} "
+                "and 'abstract_chars_count' set to 0. The feed won't be compliant, "
+                "because an item must have a description."
+            )
```

Comment:
Thanks everyone working on this!

I think both doc and code are much clearer now, except this warning.

- `abstract_chars_count` may not be set to 0. It also goes here if `abstract_chars_count: -10`.
- “No _description_ set” is a bit ambiguous. `<description>` in RSS or `description` in page's meta?

So I suggest the following.

```suggestion
            logger.warning(
                f"No description generated from metadata or content of the page {in_page.file.src_uri}, "
                "therefore the feed won't be compliant, "
                "because an item must have a description."
            )
```